### PR TITLE
fix: Minor bugs in deploy-cluster-capz.sh

### PR DESCRIPTION
**Pull request by ab63**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

"kind-" is repeated in the cluster name as MGMT_CLUSTER_CONTEXT already includes the prefix. Namespace name is missing while getting secret

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/deploy-cluster-capz.sh [hunk 1]
```diff
@@ -129,7 +129,7 @@ function create_workload_cluster() {
   fi
 
   echo "Waiting for the kubeconfig to become available"
-  timeout --foreground 1000 bash -c "while ! kubectl get secrets | grep ${CLUSTER_NAME}-kubeconfig; do sleep 1; done"
+  timeout --foreground 1000 bash -c "while ! kubectl get secrets -n "${CLUSTER_NAME}" | grep ${CLUSTER_NAME}-kubeconfig; do sleep 1; done"
```
Comment (lzhecheng) [inline hack/deploy-cluster-capz.sh:132]: IRRC, the secret is in default namespace. Have you check it?

Comment (ab63) [reply hack/deploy-cluster-capz.sh:132]: This was the case earlier but it was updated with this change https://github.com/kubernetes-sigs/cloud-provider-azure/commit/cd295b03129579e5834a73cb41aa78a8d1aa1898

Comment (lzhecheng) [reply hack/deploy-cluster-capz.sh:132]: Got it. I was not aware of the change.

**Comment by ab63** [comment]
The failure should be fixed with https://github.com/kubernetes-sigs/cloud-provider-azure/commit/79d3cc48c13b4fdb21b0f9ce76b3460ae8192cba

**Comment by lzhecheng** [comment]
The failure is not related to the change. Merging this PR.

**Pull request merged**